File: php/features.php
<?php
declare(strict_types=1);

// Ipify SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class IpifyFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new IpifyBaseFeature();
            case "test":
                return new IpifyTestFeature();
            default:
                return new IpifyBaseFeature();
        }
    }
}

File: py/features.py
# Ipify SDK feature factory

from feature.base_feature import IpifyBaseFeature
from feature.test_feature import IpifyTestFeature


def _make_feature(name):
    features = {
        "base": lambda: IpifyBaseFeature(),
        "test": lambda: IpifyTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
